src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1050,10 +1053,27 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
             }
 
             EmitBlock(catchBlock.Body);
-
             _builder.CloseLocalScope();
         }
 
+        private static bool IsFilterConstantFalse(BoundCatchBlock catchBlock)
+        {
+            // Depending on compilation options, BoundLiteral can be wrapped by BoundSequence or BoundSequencePointExpression
+            if(catchBlock.ExceptionFilterOpt != null)
+            {
+                if(catchBlock.ExceptionFilterOpt is BoundSequence sequence)
+                {
+                    return sequence.Value is BoundSequencePointExpression expression
+                            && expression.Expression?.ConstantValue?.BooleanValue == false;
```

Comment:
This is not correct. Sequence that returns a literal is not the same as a literal. Sequence may have observable sideeffects so one cannot be replaced with another.